src/protocol/config.c

```diff
@@ -75,20 +77,25 @@ z_result_t _z_str_intmap_from_strn(_z_str_intmap_t *strint, const char *s, uint8
 
             // Read and populate the value
             const char *p_value_start = _z_cptr_char_offset(p_key_end, 1);
-            const char *p_value_end = strchr(p_key_end, INT_STR_MAP_LIST_SEPARATOR);
+            size_t value_max_size = curr_len - _z_ptr_char_diff(p_value_start, start);
+            const char *p_value_end = memchr(p_key_end, INT_STR_MAP_LIST_SEPARATOR, value_max_size);
+
+            size_t p_value_len = 0;
             if (p_value_end == NULL) {
                 p_value_end = end;
+                p_value_len = value_max_size;
+            } else {
+                p_value_len = _z_ptr_char_diff(p_value_end, p_value_start);
             }
-
-            size_t p_value_len = _z_ptr_char_diff(p_value_end, p_value_start) + (size_t)1;
-            char *p_value = (char *)z_malloc(p_value_len);
+            char *p_value = (char *)z_malloc(p_value_len + 1);
             if (p_value != NULL) {
-                _z_str_n_copy(p_value, p_value_start, p_value_len);
```

Comment:
If we still need `\0` at the end it will be better to use strncmp (or _z_str_n_copy like before with adjusted length)